### ISSUE TITLE
sonic-pi-cli on Windows 10

### ISSUE THREAD

**Issue by rbnpi** [Widdershin/sonic-pi-cli#14]
I have recently been trying to get the gem to work with the release version of Sonic Pi 2.11 on Windows 10. SP is running fine, and the server log file says
```
Sonic Pi server booting...
Using protocol: udp
Detecting port numbers...
Send port: 4558
Listen port: 4557
  - OK
Scsynth port: 4556
  - OK
Scsynth send port: 4556
  - OK
OSC cues port: 4559
  - OK
Booting server...
```
so port 4557 seems to be open ok. However when I run sonic_pi it give the error
```
Sonic Pi is not listening on 4557 - is it running?
```
Any ideas? It works perfectly on a Mac with SP 2.11

**Comment by rbnpi** [comment]
I have done some further tests with this. If I comment out the call to test the connection in line 48 of sonic_pi 
```
#app.test_connection!
```
Then it works fine. Somehow the test fails on Windows, even though it works on my Mac and RPi.

**Comment by davidair** [comment]
I think this is because test_connection is using the osc-ruby's Server to test if the server is already running. If we look at the way Server is implemented (https://github.com/aberant/osc-ruby/blob/master/lib/osc-ruby/server.rb#L5), we'll see that it uses the following line to bind the socket:

@socket.bind( '', port )

It would be probably possible to replace OSC::Server.new(port) with these lines:

`
require 'socket'

...

socket = UDPSocket.new
socket.bind(nil, port)
`
Notice the nil argument, which implies localhost (as mentioned here: https://stackoverflow.com/questions/1722512/ruby-udp-server-client-test-fails?rq=1)

**Comment by Widdershin** [comment]
Good catch @rbnpi, and thanks for the analysis @davidair.

Is someone interested in making a pull request to implement the changes @davidair proposes and testing that it works on Windows? I can check it on OSX and Linux.

**Comment by davidair** [comment]
I've created a PR and tested this on OSX - will test on Windows 10 later tonight and post the findings here.

**Comment by davidair** [comment]
Successfully tested the PR on Windows 10  (note that to make the test work, I had to change "require sonic_pi" in bin/sonic_pi to "require_relative '../lib/sonic_pi'")

**Comment by Widdershin** [comment]
Fixed by the wonderful @davidair, and that fix has been released in v0.1.1.

Thanks again for reporting this issue @rbnpi, I appreciate your continued contributions to this project.

I've added you both as contributors on Github. This doesn't obligate you to contribute in any way, but it means we can work together on branches for any future work. I also welcome you to review pull requests, triage issues, and if you desire I'll happily grant your release rights.

:smile_cat:
